src/opt/horizontal_fusion_impl.cpp

```diff
@@ -0,0 +1,848 @@
+#include "horizontal_fusion_impl.hpp"
+#include <migraphx/iterator_for.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+static unsigned opcode_bits  = 16;
+static unsigned hash_id_bits = 16;
+static unsigned filter_bits  = 8;
+static unsigned kernel_bits  = 8;
+
+std::unordered_map<std::string, encoder> horizontal_fusion_impl::op_registry =
```

Comment:
This wont work, `-Wglobal-constructors` will complain. The should be a local static variable that is initialized in place:

```cpp
const std::unordered_map<std::string, encoder>& get_op_registery()
{
    static std::unordered_map<std::string, encoder> m = {
        {"gpu::convolution", encode_conv_common},
        {"gpu::conv_bias_relu", encode_conv_common},
        {"hip::add_relu", encode_common},
        {"convolution", encode_conv_common},
        {"add", encode_common},
        {"relu", encode_common},
    };
    return m;
}
```

The same applies to `op_flag`.

Reply:
fixed.